### PR TITLE
Dollmaster Fixes

### PR DESCRIPTION
**Pull request by D1GQ**
# Overview
More fixes for the beloved Dollmaster!
# Futures
```diff
+ Possibly fixed issue of players being swapped during meeting.
+ Fixed issue possessing player during comms camouflage sabotage.
+ Fixed desync issues with swapping players cosmetic.
+ Fixed Dollmaster color showing on map for possessed player.
+ Fixed possessed player being able to see self/dollmaster outfit inside a fungal mushroom cloud.
+ Fixed vanilla info diplay in Camouflage Comms.
```

### CODE REVIEW

**Comment by Ultradragon005** [comment]
I swear how many  bug fix prs is this role gonna have DAMN 💀

**Comment by D1GQ** [comment]
True lmao, To be fair Dollmaster is a bit more complex than most roles.

**Comment by Dxgamer7405** [comment]
I thought about something, what if the tracker tracks the dollmaster? he will probably be able to see who is controlling him that way

**Comment by D1GQ** [comment]
> I thought about something, what if the tracker tracks the dollmaster? he will probably be able to see who is controlling him that way

This shouldn't be the case due to it now keeping the PlayerNetworkedData the same. Now it will be tracking the wrong player but there's no way to fix that at the time!

File: Modules/Utils.cs [hunk 1]
```diff
@@ -1824,6 +1824,13 @@ static int GetInfoSize(string RoleInfo)
                         SelfName = GetString("DevouredName");
                 }
 
+                // Dollmaster, Prevent seeing self in mushroom cloud
+                if (CustomRoles.DollMaster.HasEnabled() && seerRole != CustomRoles.DollMaster)
+                {
+                    if (DollMaster.IsDoll(seer.PlayerId))
+                        SelfName = "<size=10000%><color=#000000>■</size></color>";
```
Comment (Tommy-XL) [inline Modules/Utils.cs:1831]: html code is written incorrectly
Should be:
`"<size=10000%><color=#000000>■</color></size>"`

File: Patches/PlayerControlPatch.cs [hunk 2]
```diff
@@ -1325,6 +1325,13 @@ public static Task DoPostfix(PlayerControl __instance)
                         RealName = GetString("DevouredName");
                 }
 
+                // Dollmaster, Prevent seeing self in mushroom cloud
+                if (CustomRoles.DollMaster.HasEnabled() && seerRole != CustomRoles.DollMaster)
+                {
+                    if (DollMaster.IsDoll(seer.PlayerId))
+                        RealName = "<size=10000%><color=#000000>■</size></color>";
```
Comment (Tommy-XL) [inline Patches/PlayerControlPatch.cs:1332]: Same

File: Modules/Utils.cs (hunk 1 above)
Comment (D1GQ) [reply Modules/Utils.cs:1831]: I'll fix it, but how's that incorrect? I've always written html code like that in among us and never had an issue.

Comment (Ultradragon005) [reply Modules/Utils.cs]: In the provided snippet, it works as expected. it's just not good practice to begin closing the outer-most tags BEFORE closing the inner-most tags
image

**Pull request merged**